lib/src/component/error_boundary.dart

```diff
@@ -183,4 +250,136 @@ class ErrorBoundaryComponent<T extends ErrorBoundaryProps, S extends ErrorBounda
       throw new PropError.value(children, 'children', 'ErrorBoundary accepts only a single ReactComponent child.');
     }
   }
+
+  /// Resets the [ErrorBoundary] to a non-error state.
+  ///
+  /// This can be called manually on the component instance using a `ref` -
+  /// or by passing in a new child instance after a child has thrown an error.
+  void reset() {
+    setState((newState()
+      ..hasError = false
+      ..showFallbackUIOnError = props.fallbackUIRenderer != null
+    ), _resetInternalErrorTracking);
```

Comment:
Why is this being done after the state change as opposed to before it?

The case that comes to mind is the reset call from `componentDidUpdate`... I feel like we should start with a clean slate whenever the child changes.

Reply:
Addressed in https://github.com/Workiva/over_react/pull/350/commits/1998d8daf855f671b382255e5978a0fb41130428